diff --git a/adb/types.h b/adb/types.h
index deca7eafb..97abc41aa 100644
--- a/adb/types.h
+++ b/adb/types.h
@@ -23,6 +23,8 @@
 #include <utility>
 #include <vector>
 
+#include <string.h>
+
 #include <android-base/logging.h>
 
 #include "fdevent/fdevent.h"
diff --git a/base/cmsg.cpp b/base/cmsg.cpp
index 1fa873c82..a3007ce5c 100644
--- a/base/cmsg.cpp
+++ b/base/cmsg.cpp
@@ -33,7 +33,7 @@ ssize_t SendFileDescriptorVector(borrowed_fd sockfd, const void* data, size_t le
                                  const std::vector<int>& fds) {
   size_t cmsg_space = CMSG_SPACE(sizeof(int) * fds.size());
   size_t cmsg_len = CMSG_LEN(sizeof(int) * fds.size());
-  if (cmsg_space >= PAGE_SIZE) {
+  if (cmsg_space >= (size_t)sysconf(_SC_PAGE_SIZE)) {
     errno = ENOMEM;
     return -1;
   }
@@ -75,7 +75,7 @@ ssize_t ReceiveFileDescriptorVector(borrowed_fd sockfd, void* data, size_t len,
   fds->clear();
 
   size_t cmsg_space = CMSG_SPACE(sizeof(int) * max_fds);
-  if (cmsg_space >= PAGE_SIZE) {
+  if (cmsg_space >= (size_t)sysconf(_SC_PAGE_SIZE)) {
     errno = ENOMEM;
     return -1;
   }
diff --git a/libcutils/include/cutils/threads.h b/libcutils/include/cutils/threads.h
index ba4846e33..197066171 100644
--- a/libcutils/include/cutils/threads.h
+++ b/libcutils/include/cutils/threads.h
@@ -33,7 +33,7 @@ extern "C" {
 // Deprecated: use android::base::GetThreadId instead, which doesn't truncate on Mac/Windows.
 //
 
-extern pid_t gettid();
+//extern pid_t gettid();
 
 //
 // Deprecated: use `_Thread_local` in C or `thread_local` in C++.
diff --git a/libutils/Errors.cpp b/libutils/Errors.cpp
index 74f3befac..010b53b69 100644
--- a/libutils/Errors.cpp
+++ b/libutils/Errors.cpp
@@ -13,6 +13,8 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+#include <string.h>
+
 #include <utils/Errors.h>
 
 namespace android {
